### PR TITLE
Fix NetBSD build

### PR DESCRIPTION
**Pull request by g0mb4**
Using NetBSD 10.0.

image

Compiling with: `cmake -DBUILD_IMLIB2=OFF ..`

There are a lot of `TODO`s, but it compiles.

### CODE REVIEW

**Comment by g0mb4** [comment]
Sure. I'll do the rebasing, just wanted to save my work and signal that I'm working on it.

**Comment by g0mb4** [comment]
And sorry for the inconvinience, but now I have time to work on it. I do not mind rebasing and fixing the conflicts.

**Comment by Caellian** [comment]
> And sorry for the inconvinience.

Not an inconvenience. I just warned you because I did the same thing in one of my first commits and it ended up being such a mess that I had to abandon the PR and create a new one with no history (just one example).

Anyway, merged the other branch and rebased this one (reminder to pull before making the next commit), only a single conflict.

File: cmake/ConkyPlatformChecks.cmake [hunk 1]
```diff
@@ -97,6 +97,7 @@ endif(CMAKE_SYSTEM_NAME MATCHES "SunOS")
 
 if(CMAKE_SYSTEM_NAME MATCHES "NetBSD")
   set(OS_NETBSD true)
+  set(conky_libs ${conky_libs} -lkvm -lintl -lossaudio)
```
Comment (Caellian) [inline cmake/ConkyPlatformChecks.cmake:100]: - `-lintl` not needed.
- `-lossaudio` should be handled  on line 159 in this file presumably by adding `OR OS_NETBSD` to `elseif(OS_OPENBSD)`

File: cmake/ConkyPlatformChecks.cmake [hunk 2]
```diff
@@ -256,7 +261,12 @@ if(BUILD_NCURSES)
   endif(NOT CURSES_FOUND)
 
   set(conky_libs ${conky_libs} ${CURSES_LIBRARIES})
-  set(conky_includes ${conky_includes} ${CURSES_INCLUDE_DIR})
+  if(OS_NETBSD)
+    set(conky_includes ${conky_includes} /usr/pkg/include)
+    set(conky_includes ${conky_includes} /usr/pkg/include/ncurses)
+  else(OS_NETBSD)
+    set(conky_includes ${conky_includes} ${CURSES_INCLUDE_DIR})
+  endif(OS_NETBSD)
```
Comment (Caellian) [inline cmake/ConkyPlatformChecks.cmake:269]: This whole `BUILD_NCURSES` block should really do what [this patch for weechat on NetBSD does](http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/chat/weechat/patches/patch-cmake_FindNcurses.cmake?rev=1.1&content-type=text/x-cvsweb-markup) - check for both `ncurses` and `curses`.

**Review by Caellian — CHANGES_REQUESTED** [review]
Minor change requests

File: cmake/ConkyPlatformChecks.cmake [hunk 3]
```diff
@@ -249,14 +253,39 @@ endif(BUILD_HTTP)
...
+  find_path(NCURSES_INCLUDE_PATH
+    NAMES ncurses.h curses.h
+    PATH_SUFFIXES ncurses
+    PATHS /usr/include /usr/local/include /usr/pkg/include
+  )
 
-  set(conky_libs ${conky_libs} ${CURSES_LIBRARIES})
-  set(conky_includes ${conky_includes} ${CURSES_INCLUDE_DIR})
+  find_library(NCURSES_LIBRARY
+    NAMES ncurses
```
Comment (Caellian) [inline cmake/ConkyPlatformChecks.cmake:264]: Replace with `curses`. `ncurses` is open-source distribution of `curses`, `ncurses` must provide `curses` (or it's an error made by package author)

File: cmake/ConkyPlatformChecks.cmake [hunk 4]
```diff
@@ -249,14 +253,39 @@ endif(BUILD_HTTP)
 
 if(BUILD_NCURSES)
   set(CURSES_NEED_NCURSES TRUE)
-  include(FindCurses)
 
-  if(NOT CURSES_FOUND)
-    message(FATAL_ERROR "Unable to find ncurses library")
-  endif(NOT CURSES_FOUND)
+  find_path(NCURSES_INCLUDE_PATH
+    NAMES ncurses.h curses.h
```
Comment (Caellian) [inline cmake/ConkyPlatformChecks.cmake:258]: Check only for `curses.h`, we want to be evil and fail here if `ncurses.h` exists and `curses.h` re-export doesn't (it shouldn't; checked on Arch, NetBDS diff also replaces it with `curses.h`).

**Review by Caellian — CHANGES_REQUESTED** [review]

**Review by Caellian — APPROVED** [review]
CI still builds for other platforms, LGTM.

**Pull request merged**